### PR TITLE
Add mention of Composer and PHP compatibility to project README.

### LINKED ISSUE #129: Plugin is [NOT] incompatible with PHP 8
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - dealerdirect/phpcodesniffer-composer-installer v0.6.2 requires php ^5.3|^7 -> your PHP version (8.0.0) does not satisfy that requirement.
    - dealerdirect/phpcodesniffer-composer-installer v0.6.1 requires php ^5.3|^7 -> your PHP version (8.0.0) does not satisfy that requirement.
    - dealerdirect/phpcodesniffer-composer-installer v0.6.0 requires php ^5.3|^7 -> your PHP version (8.0.0) does not satisfy that requirement.
    - dealerdirect/phpcodesniffer-composer-installer v0.5.0 requires php ^5.3|^7 -> your PHP version (8.0.0) does not satisfy that requirement.
```

How is this invalid?

This suddenly broke a CI for me. 🤷‍♀️

_Originally posted by @lilithebowman in https://github.com/Dealerdirect/phpcodesniffer-composer-installer/issues/119#issuecomment-745424644_

### PR DESCRIPTION
**Pull request by Potherca**
Fixes #129

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +27,14 @@ composer require --dev dealerdirect/phpcodesniffer-composer-installer
 
 That's it.
 
+### Compatibility
+
+This plugin is compatible with both version **1.x** and **2.x** of [Composer][composer], and both version **2.x** and **3.x** of [PHP_CodeSniffer][codesniffer].
+
+PHP **5.x**, **7.x**, and **8.x** are all supported.
+
+For PHP 8 version `0.7.x` (or higher) of this plugin must be _explicitly_ set as  version constraints, as Composer treats minors releases below 1.0 as major releases. In other words: using `^0.6` will not allow Composer to install v0.7 and thus _not_ support PHP8.
```
Comment (jrfnl) [inline README.md:36]: This goes for _both_ Composer 2.x as well as PHP 8.0.

Suggestion for alternative content of this section:

> This plugin is compatible with:
> * PHP 5.x, 7.x and 8.x.
>     Support for PHP 8.0 was added in `v0.7.0`.
> * [Composer][composer] 1.x and 2.x.
>     Support for Composer 2.x was added in `v0.7.0`.
> * [PHP_CodeSniffer][codesniffer] 2.x and 3.x.
>     Support for PHP_CodeSniffer 3.x was added in `v0.4.0`.
>
> Note: as Composer treats minor releases below 1.0.0 as major releases, version `0.7.x` (or higher) of this plugin must be _explicitly_ set as version constraint when using Composer 2.x or PHP 8.0.
> In other words: using `^0.6` will not work with Composer 2.x or PHP 8.0.

Comment (Potherca) [reply README.md:36]: Will amend.

**Comment by Potherca** [comment]
@jrfnl  I finally found some time to update this! :tada: 

I've changed the wording, as you suggested. Some things to note:

- I added links to various things, just in case people want to know more
- I placed the comment in a block (using `>`) to make it similar to the warning block further down in the document
- I added an emoji as icon for the block, to make it stand out more (and look similar to the other block). I didn't feel this message warranted a "warning", so I opted for "Information" instead.

File: README.md [hunk 2]
```diff
@@ -29,6 +27,17 @@ composer require --dev dealerdirect/phpcodesniffer-composer-installer
 
 That's it.
 
+### Compatibility
+
+This plugin is compatible with:
+
+- PHP **5.x**, **7.x**, and **8.x** (Support for PHP v8 is available since [`v0.7.0`][v0.7])
```
Comment (jrfnl) [inline README.md:34]: Just placing a note here for the record:
While we currently can only guarantee support with PHP 8.0, the current test build against PHP 8.1 _does_ seem to pass, so I'm fine with leaving this at `v8`.

**Review by jrfnl — APPROVED** [review]

**Pull request merged**